test/hotspot/jtreg/vmTestbase/nsk/share/jpda/SocketIOPipe.java

```diff
@@ -219,23 +270,17 @@ protected void connect() {
         if (shouldStop)
             return;
 
-        connection = new SocketConnection(this, getName());
-
         if (listening) {
-            connection.setConnectingProcess(connectingProcess);
-            if (serverSocket == null) {
-                connection.bind(port, timeout);
-            } else {
-                connection.setServerSocket(serverSocket);
+            // listenerThread == null means the test is not updated yet
+            // to start IOPipe listening before launching debuggee.
+            if (listenerThread == null) {
+                // start listening and accept connection on the current thread
+                listenerThread = new ListenerThread();
+                listenerThread.run();
```

Comment:
It is not clear, why `listenerThread.run()` is used instead of `listenerThread.start()` as at the line 257.

Reply:
This is part of the workaround for non-updated tests - the tests should work the same way they did.
start() would start new thread, so listening starts a bit later and we have more chances to get "connection refused" error.